mbozee/session05/mailroom.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env python3
+
+
+def menu():
+    """Print out list of options."""
+    print("menu>>   <name>:add donor and donation   l:list donors   r:print report   t:thank you   h:help   q:quit")
+
+donors = {
+    'Donovan Leitch': [100, 20, 1000],
+    'Stevie Nicks': [200, 500],
+    'Ian Anderson': [190],
+    'Mavis Staples': [170, 40, 130],
+    'Suzanne Vega': [160, 300]
+}
+
+
+def thank_you(name, value):
+    """Compose and print thank you letter."""
+    letter = '''
+    Dear {},
+
+    Thank you for your generous donation of ${}.
+    Your gift makes a major impact on aspiring musicians in the Seattle area.
+
+    Sincerely,
+    Seattle Music Fund
+    '''.format(name, value)
+    print(letter)
+
+
+def create_report():
+    """Generate a report of donors and donation amounts."""
+    donors_sorted = sorted(donors.items(),
+                           key=lambda t: sum(t[1]), reverse=True)
+
+    print("Donor Name                | Total Given | Num Gifts | Avg Gift")
+
+    for donor in donors_sorted:
+        print("{} | {} | {} | {}".format(
+            donor[0].ljust(25),
+            str(sum(donor[1])).rjust(11),
+            str(len(donor[1])).rjust(9),
+            str(round(sum(donor[1])/len(donor[1]))).rjust(8)
+        ))
+
+    print('\n', end='')
+    menu()
+
+
+def list():
+    """Print a list of donor names."""
+    for donor in donors:
+        print(donor)
+
+
+def help():
```

Comment:
`help` is also a standard function, probably want to rename this to avoid weirdness.